### PR TITLE
[CLC-215]: Serializer Generator Documentation

### PR DESCRIPTION
**Pull request by kutluhanmetin**

### CODE REVIEW

File: docs/modules/ROOT/pages/clc-serializer-generator.adoc [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+# All objects in this file will share the same namespace.
+classes:
+ - name: <name of the class>
+   fields:
+     - name: <field name>
+       type: <field type>
+       external: bool # to mark external types (external to this yaml file)
+----
+
+==== namespace:
```
Comment (yuce) [inline docs/modules/ROOT/pages/clc-serializer-generator.adoc:80]: Could you add a blank line after headers and remove the oclumns?

File: docs/modules/ROOT/pages/clc-serializer-generator.adoc [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+Used for logical grouping of classes. Typically, for every namespace, you will have a schema file. Namespace is optional. If not provided, the classes will be generated at global namespace (no namespace). The user should provide the language specific best practice when using the namespace. The tool will use the namespace while generating code if provided.
+
+==== imports:
+Used to import other schema files. The type definitions in the imported yaml schemas can be used within this yaml file. Cyclic imports will be checked and handled properly. For this version of the tool, an import can only be a single file name and the tool will assume all yaml files imported will be in the same directory as the importing schema file.
+
+==== classes:
+Used to define classes in the schema.
+
+* name
```
Comment (yuce) [inline docs/modules/ROOT/pages/clc-serializer-generator.adoc:89]:
```suggestion
* name: Name of the class
```

File: docs/modules/ROOT/pages/clc-serializer-generator.adoc [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+* fields
```
Comment (yuce) [inline docs/modules/ROOT/pages/clc-serializer-generator.adoc:90]:
```suggestion
* fields: Field specification of the class
```

**Review by yuce — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/clc-serializer-generator.adoc (hunk 1 above)
Comment (kutluhanmetin) [reply docs/modules/ROOT/pages/clc-serializer-generator.adoc]: https://github.com/hazelcast/hazelcast-commandline-client/pull/406/commits/29ec891eadaca81a0a738584a071ccf8072acab7

File: docs/modules/ROOT/pages/clc-serializer-generator.adoc (hunk 2 above)
Comment (kutluhanmetin) [reply docs/modules/ROOT/pages/clc-serializer-generator.adoc]: https://github.com/hazelcast/hazelcast-commandline-client/pull/406/commits/29ec891eadaca81a0a738584a071ccf8072acab7

File: docs/modules/ROOT/pages/clc-serializer-generator.adoc (hunk 3 above)
Comment (kutluhanmetin) [reply docs/modules/ROOT/pages/clc-serializer-generator.adoc]: https://github.com/hazelcast/hazelcast-commandline-client/pull/406/commits/29ec891eadaca81a0a738584a071ccf8072acab7

**Pull request merged**